test/e2e/testsuites/failed_mount.go

```diff
@@ -151,10 +151,6 @@ func (t *gcsFuseCSIFailedMountTestSuite) DefineTests(driver storageframework.Tes
 
 		ginkgo.By("Deleting the Kubernetes service account")
 		testK8sSA.Cleanup(ctx)
-
-		ginkgo.By("Checking that the pod has failed mount error Unauthenticated")
```

Comment:
Shouldn't there still be some auth error?

Reply:
This check was emerging from the storage client creation [here](https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver/blob/v1.3.1/pkg/csi_driver/node.go#L143) which has been removed. Also GCS Fuse does a one time bucket check during first attempt of mount and reports failure. Node republish get that error.

here is how the pod events look for this test case:
```
Events:
  Type     Reason       Age               From               Message
  ----     ------       ----              ----               -------
  Normal   Scheduled    14s               default-scheduler  Successfully assigned failed-mount-178/gcsfuse-volume-tester-bbc9w to gke-test-cluster-1-29-pool-1-1034925b-9nmt
  Normal   Pulled       13s               kubelet            Container image "gcr.io/saikatroyc-stateful-joonix/gcs-fuse-csi-driver-sidecar-mounter:test-v1" already present on machine
  Normal   Created      13s               kubelet            Created container gke-gcsfuse-sidecar
  Normal   Started      13s               kubelet            Started container gke-gcsfuse-sidecar
  Warning  Failed       12s               kubelet            Error: failed to generate container "cf52d1e76c3168b879fa9fd5ea78c39d00528d14deea4eea7d99669c0755eca9" spec: failed to generate spec: failed to stat "/var/lib/kubelet/pods/a4291a10-4d68-4db3-aa05-722442b92188/volumes/kubernetes.io~csi/gcsfuse.csi.storage.gke.io-2x54n/mount": stat /var/lib/kubelet/pods/a4291a10-4d68-4db3-aa05-722442b92188/volumes/kubernetes.io~csi/gcsfuse.csi.storage.gke.io-2x54n/mount: transport endpoint is not connected
  Normal   Pulled       1s (x2 over 13s)  kubelet            Container image "registry.k8s.io/e2e-test-images/busybox:1.36.1-1" already present on machine
  Warning  Failed       1s                kubelet            Error: failed to generate container "5399d89f9091190c86f9314461a71af60015dfddad9af6fd78190f17e0465ec5" spec: failed to generate spec: failed to stat "/var/lib/kubelet/pods/a4291a10-4d68-4db3-aa05-722442b92188/volumes/kubernetes.io~csi/gcsfuse.csi.storage.gke.io-2x54n/mount": stat /var/lib/kubelet/pods/a4291a10-4d68-4db3-aa05-722442b92188/volumes/kubernetes.io~csi/gcsfuse.csi.storage.gke.io-2x54n/mount: transport endpoint is not connected
  Warning  FailedMount  0s (x2 over 1s)   kubelet            MountVolume.SetUp failed for volume "gcsfuse.csi.storage.gke.io-2x54n" : rpc error: code = PermissionDenied desc = the sidecar container failed with error: Error while mounting gcsfuse: mountWithArgs: mountWithStorageHandle: fs.NewServer: create file system: SetUpBucket: Error in iterating through objects: googleapi: Error 403: Caller does not have storage.objects.list access to the Google Cloud Storage bucket. Permission 'storage.objects.list' denied on resource (or it may not exist)., forbidden
gcsfuse exited with error: exit status 1
```